### PR TITLE
upgrade the go libraries to latest version

### PR DESCRIPTION
**Pull request by haoqing0110**
Ref: https://github.com/stolostron/backlog/issues/23893
Signed-off-by: haoqing0110 <qhao@redhat.com>

### CODE REVIEW

File: overlays/deploy/deployment.yaml [hunk 1]
```diff
@@ -17,4 +17,4 @@ spec:
       volumes:
       - name: certs
         secret:
-          secretName: clusterlifecycle-state-metrics-certs
+          secretName: clusterlifecycle-state-metrics
```
Comment (xuezhaojun) [inline overlays/deploy/deployment.yaml:20]: Where is this secret from?

Comment (haoqing0110) [reply overlays/deploy/deployment.yaml:20]: @xuezhaojun it should be created by k8s when creating `service`

Comment (xuezhaojun) [reply overlays/deploy/deployment.yaml:20]: Got it.

**Pull request merged**